### PR TITLE
Add logentries engine

### PR DESCRIPTION
**Pull request by jcftang**
This engine is first attempt at a simple engine to send event information
to the Logentries logging service.

### CODE REVIEW

File: salt/engines/logentries.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            transport=__opts__['transport'],
+            opts=__opts__,
+            sock_dir=__opts__['sock_dir'],
+            listen=True)
+    log.debug('Logentries engine started')
+
+    try:
+        val = uuid.UUID(token)
+    except ValueError:
+        log.warning('Not a valid token'.format(token))
```
Comment (cachedout) [inline salt/engines/logentries.py]: Did you forget a `{0}` in here?

**Comment by cachedout** [comment]
@jcftang This looks really nice! I think there's one possible typo in there but if you could take a look at that please then I think we can get this merged. Nice work!

File: salt/engines/logentries.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+'''
+An engine that sends events to the Logentries logging service.
+
+:maintainer:  Jimmy Tang (jimmy_tang@rapid7.com)
+:maturity:    New
+:depends:     ssl, certifi
+:platform:    all
+
+.. versionadded: 2015.8.1
```
Comment (jfindlay) [inline salt/engines/logentries.py]: @jcftang, this needs to be `.. versionadded:: Boron`.

**Comment by cachedout** [comment]
@jcftang I have just gone ahead and made the necessary changes in #29284, preserving your commits as well. So, your work has now been merged. Thank you!

File: salt/engines/logentries.py (hunk 1 above)
Comment (jcftang) [inline salt/engines/logentries.py]: Yes I did forget that, my mistake

**Comment by jcftang** [comment]
@cachedout thanks for the merge, I missed the mails earlier on relating to the required changes (was out of the office). Please let me know if more modifications are needed, I'll be happy to fix it up as needed.

**Pull request merged**